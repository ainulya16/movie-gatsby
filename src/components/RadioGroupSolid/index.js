import React from 'react';
import { Label } from '@aia-digital/ui-library';
import { Radio } from 'antd';
import styled from 'styled-components';
import { string, arrayOf, bool, func, shape } from 'prop-types';

const DefaultWrapper = styled.div`
  .ant-radio-group-solid
    .ant-radio-button-wrapper-checked:not(.ant-radio-button-wrapper-disabled) {
    background: #d31145;
    border-color: #d31145;
  }
  .ant-radio-group-solid
    .ant-radio-button-wrapper-checked:not(.ant-radio-button-wrapper-disabled):hover {
    background: #d31145;
    border-color: #d31145;
    box-shadow: -1px 0 0 0 #d31145;
  }
  .ant-radio-button-wrapper {
    border-radius: unset !important;
    width: 150px;
    text-align: center;
  }

  .ant-radio-button-wrapper-checked {
    box-shadow: -1px 0 0 0 #d31145;
  }

  .ant-radio-button-wrapper {
    margin-right: 10px;
    font-family: 'AIABody', Arial, 'Helvetica Neue', Helvetica, sans-serif;
    font-size: 15px;
  }
  .ant-radio-button-wrapper:hover {
    color: rgba(0, 0, 0, 0.65);
  }
`;

const RadioGroupSolid = props => {
  const { label, data, onChange, defaultValue, name } = props;
  return (
    <DefaultWrapper className="radioGroupWrapper">
      {label.length > 0 && <Label>{label}</Label>}
      <Radio.Group
        defaultValue={defaultValue}
        buttonStyle="solid"
        onChange={onChange}
        name={name}
      >
        {data.length > 0 &&
          data.map(item => (
            <Radio.Button key={data.indexOf(item)} value={item.value}>
              {item.text}
            </Radio.Button>
          ))}
      </Radio.Group>
    </DefaultWrapper>
  );
};

RadioGroupSolid.propTypes = {
  label: string,
  data: arrayOf(shape({ value: string, text: string, disabled: bool }))
    .isRequired,
  onChange: func.isRequired,
  name: string.isRequired,
  defaultValue: string,
};
RadioGroupSolid.defaultProps = {
  label: '',
  defaultValue: '',
};

export default RadioGroupSolid;
