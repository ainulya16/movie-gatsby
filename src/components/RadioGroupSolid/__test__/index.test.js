import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import RadioGroupSolid from '../index';

let component;
const data = [
  { value: 'electronic', text: 'Electronic', disabled: false },
  { value: 'hardCopy', text: 'Hardcopy', disabled: false },
];
const name = 'a';
const onChange = jest.fn();

describe('<RadioGroupSolid />', () => {
  beforeEach(() => {
    component = mount(
      <RadioGroupSolid data={data} name={name} onChange={onChange} />,
    );
  });
  it('should match with snapshot', () => {
    const tree = renderer
      .create(<RadioGroupSolid data={data} name={name} onChange={onChange} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should have prop data', () => {
    expect(component.prop('data')).toEqual(data);
  });
  it('should have prop name', () => {
    expect(component.prop('name')).toEqual(name);
  });
  it('should have onChange function', () => {
    expect(component.prop('onChange')).toEqual(onChange);
  });
});
