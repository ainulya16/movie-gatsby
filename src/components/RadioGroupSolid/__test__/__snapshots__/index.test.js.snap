// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroupSolid /> should match with snapshot 1`] = `
.c0 .ant-radio-group-solid .ant-radio-button-wrapper-checked:not(.ant-radio-button-wrapper-disabled) {
  background: #d31145;
  border-color: #d31145;
}

.c0 .ant-radio-group-solid .ant-radio-button-wrapper-checked:not(.ant-radio-button-wrapper-disabled):hover {
  background: #d31145;
  border-color: #d31145;
  box-shadow: -1px 0 0 0 #d31145;
}

.c0 .ant-radio-button-wrapper {
  border-radius: unset !important;
  width: 150px;
  text-align: center;
}

.c0 .ant-radio-button-wrapper-checked {
  box-shadow: -1px 0 0 0 #d31145;
}

.c0 .ant-radio-button-wrapper {
  margin-right: 10px;
  font-family: 'AIABody',Arial,'Helvetica Neue',Helvetica,sans-serif;
  font-size: 15px;
}

.c0 .ant-radio-button-wrapper:hover {
  color: rgba(0,0,0,0.65);
}

<div
  className="radioGroupWrapper c0"
>
  <div
    className="ant-radio-group ant-radio-group-solid"
  >
    <label
      className="ant-radio-button-wrapper"
    >
      <span
        className="ant-radio-button"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-button-input"
          name="a"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="electronic"
        />
        <span
          className="ant-radio-button-inner"
        />
      </span>
      <span>
        Electronic
      </span>
    </label>
    <label
      className="ant-radio-button-wrapper"
    >
      <span
        className="ant-radio-button"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-button-input"
          name="a"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="hardCopy"
        />
        <span
          className="ant-radio-button-inner"
        />
      </span>
      <span>
        Hardcopy
      </span>
    </label>
  </div>
</div>
`;
